internal/services/oracle/autonomous_database_regular_resource.go

```diff
@@ -0,0 +1,410 @@
+// Copyright © 2024, Oracle and/or its affiliates. All rights reserved
+
+package oracle
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/oracledatabase/2024-06-01/autonomousdatabases"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/oracle/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+var _ sdk.Resource = AutonomousDatabaseRegularResource{}
+
+type AutonomousDatabaseRegularResource struct{}
+
+type AutonomousDatabaseRegularResourceModel struct {
+	Location          string            `tfschema:"location"`
+	Name              string            `tfschema:"name"`
+	ResourceGroupName string            `tfschema:"resource_group_name"`
+	Tags              map[string]string `tfschema:"tags"`
+
+	// Required
+	AdminPassword                string  `tfschema:"admin_password"`
+	BackupRetentionPeriodInDays  int64   `tfschema:"backup_retention_period_in_days"`
+	CharacterSet                 string  `tfschema:"character_set"`
+	ComputeCount                 float64 `tfschema:"compute_count"`
+	ComputeModel                 string  `tfschema:"compute_model"`
+	DataStorageSizeInTbs         int64   `tfschema:"data_storage_size_in_tbs"`
+	DbVersion                    string  `tfschema:"db_version"`
+	DbWorkload                   string  `tfschema:"db_workload"`
+	DisplayName                  string  `tfschema:"display_name"`
+	LicenseModel                 string  `tfschema:"license_model"`
+	AutoScalingEnabled           bool    `tfschema:"auto_scaling_enabled"`
+	AutoScalingForStorageEnabled bool    `tfschema:"auto_scaling_for_storage_enabled"`
+	MtlsConnectionRequired       bool    `tfschema:"mtls_connection_required"`
+	NcharacterSet                string  `tfschema:"ncharacter_set"`
+	SubnetId                     string  `tfschema:"subnet_id"`
+	VnetId                       string  `tfschema:"vnet_id"`
+
+	// Optional
+	CustomerContacts []string `tfschema:"customer_contacts"`
+}
+
+func (AutonomousDatabaseRegularResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.Location(),
+
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.AutonomousDatabaseName,
+			ForceNew:     true,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		// Required
+		"admin_password": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			Sensitive:    true,
+			ForceNew:     true,
+			ValidateFunc: validate.AutonomousDatabasePassword,
+		},
+
+		"backup_retention_period_in_days": {
+			Type:     pluginsdk.TypeInt,
+			Required: true,
+			ForceNew: true,
+		},
```

Comment:
please validate this, i assume something like -1000 or 9999999 is not valid?

Reply:
Done.